src/export.cc

```diff
@@ -138,18 +138,22 @@ namespace Export {
 
 ExportMesh::ExportMesh(const PolySet& ps)
 {
+  std::map<Vertex, int> vertexMap;
   std::vector<std::array<int, 3>> triangleIndices;
   for (const auto& p : ps.polygons) {
-    auto pos1 = vertexMap.emplace(std::make_pair<std::array<double, 3>, int>({p[0].x(), p[0].y(), p[0].z()}, vertexMap.size()));
-    auto pos2 = vertexMap.emplace(std::make_pair<std::array<double, 3>, int>({p[1].x(), p[1].y(), p[1].z()}, vertexMap.size()));
-    auto pos3 = vertexMap.emplace(std::make_pair<std::array<double, 3>, int>({p[2].x(), p[2].y(), p[2].z()}, vertexMap.size()));
+    auto pos1 = vertexMap.emplace(std::make_pair<Vertex, int>({p[0].x(), p[0].y(), p[0].z()}, vertexMap.size()));
+    auto pos2 = vertexMap.emplace(std::make_pair<Vertex, int>({p[1].x(), p[1].y(), p[1].z()}, vertexMap.size()));
+    auto pos3 = vertexMap.emplace(std::make_pair<Vertex, int>({p[2].x(), p[2].y(), p[2].z()}, vertexMap.size()));
     triangleIndices.push_back({pos1.first->second, pos2.first->second, pos3.first->second});
   }
 
-  int index = 0;
-  std::map<int, int> indexTranslationMap;
+  std::vector<size_t> indexTranslationMap(vertexMap.size());
+  vertices.reserve(vertexMap.size());
+
+  size_t index = 0;
   for (const auto& e : vertexMap) {
-    indexTranslationMap.emplace(e.second, index++);
+    vertices.push_back(e.first);
+    indexTranslationMap[e.second] = index++;
   }
 
   for (const auto& i : triangleIndices) {
```

Comment:
Technically, we might want to rotate the indices of the triangle *after* the remapping to ensure the first index comes first, in case some mesh algorithm rotates our faces arbitrarily. Can do that later if that comes to be an issue.